azure-pipelines.yml

```diff
@@ -23,31 +22,15 @@ jobs:
   displayName: Build and Test Sources
   pool: Hosted VS2017
 
+  variables:
+    ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest')) }}:
+      _OfficialBuildIdArgs: /p:OfficialBuildId=$(BUILD.BUILDNUMBER)
+
   steps:
   - task: DotNetCoreInstaller@0
```

Comment:
Do we still need to install the DotNetCoreInstaller? if you call build.cmd with the -ci flag, when installing the SDK stated on the global.json, it will add it to the path using VSO variables. So once you've restored it, you can call DotNetCoreCLI tasks without installing it manually.

Reply:
Nice! I didn't know this, and it worked out perfectly.

I verified that each of the test commands are executed with `.dotnet\dotnet.exe` from the repo:

```
[command]D:\a\1\s\.dotnet\dotnet.exe test D:\a\1\s\src\csharp\Microsoft.Spark.UnitTest\Microsoft.Spark.UnitTest.csproj --logger trx --results-directory D:\a\_temp --configuration Release /p:CollectCoverage=true /p:CoverletOutputFormat=cobertura
```